pkg/sources/redisstreams/redisstream.go

```diff
@@ -164,35 +169,31 @@ func New(
 
 	// function which takes the messages that have been read from the Stream and turns them into ReadMessages
 	redisStreamsSource.XStreamToMessages = func(xstreams []redis.XStream, messages []*isb.ReadMessage, labels map[string]string) ([]*isb.ReadMessage, error) {
-		// for each XMessage in []XStream
-		for _, xstream := range xstreams {
+		if len(xstreams) > 1 {
+			redisStreamsSource.Log.Warnf("redisStreamsSource shouldn't have more than one Stream; xstreams=%+v", xstreams)
+		} else if len(xstreams) == 1 {
+			xstream := xstreams[0]
+
 			for _, message := range xstream.Messages {
-				var readOffset = message.ID
-
-				valIndex := 0
-				for f, v := range message.Values {
-
-					// need to make sure the ID is unique: we can make it a combination of the original
-					// message ID from Redis (timestamp+sequence num) plus its ordinal in the key/value pairs
-					// contained in the message
-					id := fmt.Sprintf("%s-%d", message.ID, valIndex)
-					valIndex = valIndex + 1
-
-					isbMsg := isb.Message{
-						Header: isb.Header{
-							MessageInfo: isb.MessageInfo{EventTime: time.Now()}, //doesn't seem like Redis offers a timestamp
-							ID:          id,                                     // assumption is that this only needs to be unique for this source vertex
-							Keys:        []string{f},
-						},
-						Body: isb.Body{Payload: []byte(v.(string))},
+				var outMsg *isb.ReadMessage
+				var err error
+				if len(message.Values) > 1 {
```

Comment:
This is going to generate different behaviors when someone sends data with single or multiple key/value pairs, one output is a single string, the other one is a json. 

I believe it is possible that both single and multiple key/value pairs will happen in one streamed data, and this implementation will make troubles. I suggest we alway use json.

@vigith @juliev0

Reply:
It's a good point. Here's what I'm thinking: how about for consistency we always Key on the first Key, and always use JSON as well?

Reply:
We do `keys []string` now, so all the keys can be there.

Reply:
oh, that's the "multi-key" feature I guess? I haven't looked at how that works yet. Cool, will do that then.